Sphinx autodoc include __init__()

jngrad opened the pull request:
Fixes #2780

Show Python `__init__()` class constructors in the documentation when they have a docstring. Otherwise don't show them (default behavior), as they are redundant with the class signature. See #2780 for the discussion on PEP and the Google style guide.

jngrad committed:
Formatting

jngrad committed:
Update year

jngrad committed:
Show __init__() documentation

KaiSzuttor commented on doc/sphinx/conf.py.in:
```diff
@@ -242,8 +242,9 @@ latex_elements = {
 latex_documents = [
-  (master_doc, 'ExtensibleSimulationPackageforSoftMatterResearch.tex', 'Extensible Simulation Package for Soft Matter Research Documentation',
-   'Florian Weik', 'manual'),
+    (master_doc, 'ExtensibleSimulationPackageforSoftMatterResearch.tex',
```
I don't know who introduced this but this is the wrong name for the package

jngrad replied:
Thanks! Fixed it there and in a few other places.

jngrad committed:
Correct wrong acronym

jngrad committed:
Fix typo

pkreissl commented on doc/sphinx/conf.py.in:
```diff
@@ -287,11 +288,10 @@ man_pages = [
 # (source start file, target name, title, author,
 #  dir menu entry, description, category)
 texinfo_documents = [
-  (master_doc, 'ExtensibleSimulationPackageforSoftMatterResearch', 'Extensible\
```
Is there a reason for having the target names be inconsistent, that is lower case vs. camel case?

KaiSzuttor replied:
we don't use latex anyway...

jngrad replied:
The lowercase target name at line 277 is for man pages, which are usually case-insensitive. I'm not even sure we can generate man pages, I tried with Python2 and got a warning concerning unsupported tabular format in tables with rowspans (`"table cell spanning" not supported`). Couldn't try on Python3 because of the `PyUnicode_InternFromString` error in my environment.

jngrad committed:
Update CMake file copy operation

With configure_file() the MDA_ESP subpackage becomes a target, so any
change to its Python file will trigger a new copy operation during the
next recompilation, e.g. when rebuilding the Sphinx documentation.

jngrad committed:
Use numpydoc docstring style

Documentation of __init__() members are moved to the class doscstring.

jngrad committed:
Formatting

KaiSzuttor commented:
thanks @jngrad

fweik commented:
bors r+
